### PR TITLE
[docs] Split the 'Group & Pivot' page

### PR DESCRIPTION
**Pull request by flaviendelangle**
[Doc preview](https://deploy-preview-4441--material-ui-x.netlify.app/x/react-data-grid/row-grouping/)

This PR only focus on moving the content.
I would be in favor of not changing the content, except things that are directly linked to the split (for instance the page description).

## New URLs

- [/x/react-data-grid/group-pivot/#row-grouping](https://mui.com/x/react-data-grid/group-pivot/#row-grouping) => [/x/react-data-grid/row-grouping](https://deploy-preview-4441--material-ui-x.netlify.app/x/react-data-grid/row-grouping/)

- [/x/react-data-grid/group-pivot/#tree-data](https://mui.com/x/react-data-grid/group-pivot/#tree-data) => [/x/react-data-grid/tree-data/](https://deploy-preview-4441--material-ui-x.netlify.app/x/react-data-grid/tree-data/)

- [/x/react-data-grid/group-pivot/#master-detail](https://mui.com/x/react-data-grid/group-pivot/#master-detail) => [/x/react-data-grid/master-detail/](https://deploy-preview-4441--material-ui-x.netlify.app/x/react-data-grid/master-detail/)

- [/x/react-data-grid/group-pivot/#aggregation](https://mui.com/x/react-data-grid/group-pivot/#aggregation) => [/x/react-data-grid/aggregation](https://deploy-preview-4441--material-ui-x.netlify.app/x/react-data-grid/aggregation)

- [/x/react-data-grid/group-pivot/#pivoting](https://mui.com/x/react-data-grid/group-pivot/#pivoting) => [/x/react-data-grid/pivoting/](https://deploy-preview-4441--material-ui-x.netlify.app/x/react-data-grid/pivoting)

## New navigation

I added an icon for the feature not released yet
Maybe we want to mark the plan on the nav bar for the pro-only or premium-only feature (like AG-Grid is doing)

image

## TODO

- [x] Split the `group-pivot` page into 5 new pages (`row-grouping` / `tree-data` / `master-detail` / `aggregation` / `pivoting`)
- [x] Change the links in the MUI X repo
- [ ] Change the link in the MUI Core repo
- [ ] Add redirections ?

### CODE REVIEW

**Comment by oliviertassinari** [comment]
Ah, interesting. I didn't anticipate that we could use the subheader to break down pages into smaller ones when I worked on point 7. in https://github.com/mui/material-ui/pull/32241. I guess for later, if this side nav ever become too long.

**Comment by flaviendelangle** [comment]
@oliviertassinari I did a split on this page where the content is obvious so we can focus on the those kind of topic.
Maybe your proposal work better.

@samuelsycamore @joserodolfofreitas how do you think this "sub-pages" should be displayed on the left navigation ?

**Comment by oliviertassinari** [comment]
Another thought,

Screenshot 2022-04-13 at 13 31 00

We have:

- Row grouping is Premium 
- Tree data is Pro
- Aggregation is Premium
- Pivoting is Premium

Maybe it would help to communicate in the side nav the plan. It could avoid developers that are not in the right plan not wasting time looking at docs for features that they can't use.

Screenshot 2022-04-13 at 13 32 23

File: docs/data/data-grid/getting-started/getting-started.md [hunk 1]
```diff
@@ -198,11 +198,11 @@ The enterprise components come in two plans: Pro and Premium.
...
-| [Tree data](/x/react-data-grid/group-pivot/#tree-data)                                 |    ❌     |                 ✅                 |                     ✅                     |
-| [Master detail](/x/react-data-grid/group-pivot/#master-detail)                         |    ❌     |                 ✅                 |                     ✅                     |
-| [Grouping](/x/react-data-grid/group-pivot/#row-grouping)                               |    ❌     |                 ❌                 |                     🚧                     |
-| [Aggregation](/x/react-data-grid/group-pivot/#aggregation)                             |    ❌     |                 ❌                 |                     🚧                     |
-| [Pivoting](/x/react-data-grid/group-pivot/#pivoting)                                   |    ❌     |                 ❌                 |                     🚧                     |
+| [Tree data](/x/react-data-grid/tree-data)                                              |    ❌     |                 ✅                 |                     ✅                     |
```
Comment (cherniavskii) [inline docs/data/data-grid/getting-started/getting-started.md:201]: Having redirects would be great!

**Comment by flaviendelangle** [comment]
> Maybe it would help to communicate in the side nav the plan

Yes, I also noticed that (see PR description)
AG-Grid is doing it.
And when we will split the "Columns" page it will be even more usefull (for the Column Pinning, Column Resizing, Column Reordering, etc...)

**Comment by samuelsycamore** [comment]
I think it makes a lot of sense to mark the Pro and Premium features with their respective icons on the nav bar. It would get annoying to have to click through to multiple pages to find out which features are Pro/Premium-only.

File: docs/data/pages.ts [hunk 2]
```diff
@@ -31,7 +31,17 @@ const pages = [
       { pathname: '/x/react-data-grid/scrolling' },
       { pathname: '/x/react-data-grid/virtualization' },
       { pathname: '/x/react-data-grid/accessibility' },
-      { pathname: '/x/react-data-grid/group-pivot', title: 'Group & Pivot' },
+      {
+        pathname: '/x/react-data-grid/row-grouping',
+        title: 'Group & Pivot',
```
Comment (m4theushw) [inline docs/data/pages.ts:36]: Do we really need to group these features under "Group & Pivot"? The master detail doesn't group rows.

**Review by m4theushw — APPROVED** [review]

File: docs/data/pages.ts (hunk 2 above)
Comment (flaviendelangle) [reply docs/data/pages.ts:36]: I just took every section of the current "Group & Pivot" page, but for me, "Master Detail" should go

- in the new "Row" group of pages with pages like "Row Spanning" or "Row Reordering" or "Row Height"

or

- at the root in a standalone page (it's what AG-Grid is doing)


What do you think ?

Comment (m4theushw) [reply docs/data/pages.ts:36]: Inside the "row" group seems better, but this can be done later. This new group also affects the pricing page.

**Pull request merged**